x-pack/plugins/security/server/routes/authentication/common.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { canRedirectRequest } from '../../authentication';
+import { wrapIntoCustomErrorResponse } from '../../errors';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+import { RouteDefinitionParams } from '..';
+
+/**
+ * Defines routes that are common to various authentication mechanisms.
+ */
+export function defineCommonRoutes({ router, authc, basePath, logger }: RouteDefinitionParams) {
+  // Generate two identical routes with new and deprecated URL and issue a warning if route with deprecated URL is ever used.
+  for (const path of ['/api/security/logout', '/api/security/v1/logout']) {
+    router.get(
+      {
+        path,
+        // Allow unknown query parameters as this endpoint can be hit by the 3rd-party with any
+        // set of query string parameters (e.g. SAML/OIDC logout request parameters).
+        validate: { query: schema.object({}, { allowUnknowns: true }) },
+        options: { authRequired: false },
+      },
+      createLicensedRouteHandler(async (context, request, response) => {
+        const serverBasePath = basePath.serverBasePath;
+        if (path === '/api/security/v1/logout') {
+          logger.warn(
```

Comment:
What do you think about integrating this with the upgrade assistant in a followup PR? I think it'd be helpful for folks when upgrading to the next major version if we make it known that _something_ in their environment is relying on this deprecated endpoint.

To say this another way, I'd personally be annoyed if I ran the upgrade assistant, and it _didn't_ warn me that this would stop working if I was relying on it 😄

Reply:
That's a great idea! I'll try to figure how hard/easy it would be. I tried it once in the distant past and UA wasn't ready for things like this back then, but I believe it's much more advanced now.